src/consensus/dkg.rs

```diff
@@ -105,16 +105,17 @@ impl DkgVoter {
         keypair: &Keypair,
         dkg_key: DkgKey,
         elders_info: EldersInfo,
+        key_index: u64,
     ) -> Vec<DkgCommand> {
-        if let Some(session) = &self.session {
-            if session.dkg_key == dkg_key && !session.complete {
+        if let Some(session) = self.sessions.get(&dkg_key) {
+            if !session.complete && session.key_index >= key_index {
```

Comment:
No, I think it should be `&&`. We want to replace the existing session if it is either complete or has lower key index than the new one (because that means it's already obsolete, even if it might not be complete).

Yes, the early return means don't replace. Otherwise do.